log4j-core/src/main/java/org/apache/logging/log4j/core/config/properties/PropertiesConfigurationBuilder.java

```diff
@@ -190,6 +193,17 @@ public PropertiesConfiguration build() {
         return builder.build(false);
     }
 
+    private static void useSyntheticLevelAndAppenderRefs(final String propertyName, final Properties loggerProps, final Properties context) {
+        final String propertyValue = (String) context.remove(propertyName);
+        if (propertyValue != null) {
+            final String[] parts = propertyValue.split(",");
+            loggerProps.setProperty("level", parts[0]);
```

Comment:
We should check that `parts` is not empty to avoid an `ArrayIndexOutOfBoundsException`. If memory serves, `",".split(",")` will result in an empty array.

Reply:
I think that split would result in a two value array of two empty strings.

Reply:
It's subtle and counter-intuitive. Alas, this test passes:
```java
@Test
public void split() {
    assertThat(",".split(",")).isEmpty();
}
```

Reply:
Indeed it is! I only remembered that because I checked the javadocs for `split` before using it here. At first, I was limiting the split to 1 match, but then I added the multiple refs thing.